tests/msc2716_test.go

```diff
@@ -220,19 +220,87 @@ func TestImportHistoricalMessages(t *testing.T) {
 			validateBatchSendRes(t, as, roomID, batchSendRes, false)
 		})
 
+		t.Run("Backfill still works after many batches are imported", func(t *testing.T) {
+			t.Parallel()
```

Comment:
TODO: Move this test under the `Federation` section once https://github.com/matrix-org/complement/pull/249 merges